daemon/algod/api/server/v2/account.go

```diff
@@ -138,10 +139,22 @@ func convertGeneratedTKV(akvs *generated.TealKeyValueStore) basics.TealKeyValue
 
 	tkv := make(basics.TealKeyValue)
 	for _, kv := range *akvs {
-		tkv[kv.Key] = basics.TealValue{
+		// Decode base-64 encoded map key
+		decodedKey, err := base64.StdEncoding.DecodeString(kv.Key)
+		if err != nil {
+			decodedKey = nil
```

Comment:
if `decodedKey = nil` , do you really want to `tkv[string(decodedKey)] = ... ` ?

Reply:
Good point, failing silently is bad. I made it propagate up the error.